### ISSUE TITLE
require gem_tasks results in uninitialized constant error

### ISSUE THREAD

**Issue by YorickPeterse** [bundler/bundler#3492]
Prior to 1.9 one could add a `require 'bundler/gem_tasks'` to their Rakefile to load a set of bundler related Rake tasks. Starting with 1.9.0 this results in the following error:

```
rake aborted!
NameError: uninitialized constant Bundler
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/vendor/thor-0.19.1/lib/thor/command.rb:1:in `<top (required)>'
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/vendor/thor-0.19.1/lib/thor/base.rb:1:in `<top (required)>'
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/vendor/thor-0.19.1/lib/thor.rb:2:in `<top (required)>'
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/vendored_thor.rb:4:in `<top (required)>'
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/gem_helper.rb:1:in `<top (required)>'
/home/yorickpeterse/.gem/ruby/2.2.0/gems/bundler-1.9.0/lib/bundler/gem_tasks.rb:1:in `<top (required)>'
/tmp/test/Rakefile:1:in `<top (required)>'
LoadError: cannot load such file -- bundler/gem_tasks
/tmp/test/Rakefile:1:in `<top (required)>'
(See full trace by running task with --trace)
```

To reproduce, create a directory and add the following files to it:

```
# Rakefile
require 'bundler/gem_tasks'
```

```
# Gemfile
source 'https://rubygems.org'

gemspec
```

```
# test.gemspec
Gem::Specification.new do |gem|
  gem.name = 'test'
end
```

This should give the following structure:

```
$ ls
Gemfile  Rakefile  test.gemspec
````

Running `rake -T` will now produce the above error.

**Comment by YorickPeterse** [comment]
It seems adding a `require 'bundler'` to the Rakefile solves it, though this wasn't needed before. Was this change intended?

**Comment by andremedeiros** [comment]
That's definitely a bug, as we try to be as backwards compatible as possible.

Is there any documentation that mentions that you only need `require 'bundler/gem_tasks` somewhere? Just curious if this is a precedent.

**Comment by YorickPeterse** [comment]
@andremedeiros Not that I know of, I've been doing it for so long I forgot where I initially found it. Having said that, the require is present here, but it fails before reaching this line: https://github.com/bundler/bundler/blob/master/lib/bundler/gem_helper.rb#L2

**Comment by andremedeiros** [comment]
@YorickPeterse That's all I needed to know, thank you! I'll look into it.

**Comment by segiddins** [comment]
I've got a fix ready

**Comment by segiddins** [comment]
The fix for this shipped in 1.9.1.

**Comment by YorickPeterse** [comment]
Thanks! :smile:
